tests/testthat/test-predict.R

```diff
@@ -12,6 +12,12 @@ swiss <-
 part <- caret::createDataPartition(swiss$Catholic, .8)[[1]]
 tr <- swiss[part, ]
 te <- swiss[-part, ]
+te_newlevel <- te
+te_newlevel$Catholic[c(3, 7, 16)] <- "unobserved in training"
+te_newlevel$Catholic[5] <- "another new level"
+te_new_missing <- te
+te_new_missing$Agriculture[1:3] <- NA
+te_new_missing$Catholic[6:7] <- NA
 # Need to test combos of:
 # reg/class x d from prep_data/not x newdata from prep_data/not
 ## Except training data not prepped and newdata prepped; that shouldn't work
```

Comment:
Needing a decoder for variable names means they are probably terrible names.

Reply:
Totally true, and I hated working with this after I made it. Good call. Changed.